include/service/entities/plugin_service.js

```diff
@@ -1079,7 +1139,13 @@ module.exports = function PluginServiceModule(pb) {
                 if(!cached_plugin || !cached_plugin.details) {
                     return PluginService.loadDetailsFile(PluginService.getDetailsPath(plugin.dirName), function (err, loadedDetails) {
                         details = loadedDetails;
-                        callback(err, !!details);
+                        if (autoSync && !pb.util.isError(err) && details) {
+                            var pluginService = new PluginService({site: site});
```

Comment:
A new pluginService with the plugin's site here seemed un-optimized. I need this so that getSettingsKV() works against the PluginService's site in syncSettings. Is there a better way to handle this? One alternative I considered was changing the current PluginService's site before calling syncSettings and then restoring it after the getSettingsKV() call, but that scares me.